modules/oe_theme_content_consultation/oe_theme_content_consultation.module

```diff
@@ -91,3 +91,14 @@ function oe_theme_content_consultation_preprocess_field(&$variables, $hook): voi
     $variables['attributes']['class'][] = 'ecl-editor';
   }
 }
+
+/**
+ * Implements hook_preprocess_HOOK().
+ *
+ * Renders Publication teaser view mode inside Document reference entity.
+ */
+function oe_theme_content_consultation_preprocess_field__oe_document_reference__oe_publication(&$variables): void {
+  $publication = $variables['element'][0]['#options']['entity'];
+  $build = \Drupal::entityTypeManager()->getViewBuilder('node')->view($publication, 'teaser');
```

Comment:
There is an error on the Consultation creation page `/node/add/oe_consultation` when document reference with Publication is added:
```
TypeError: Argument 1 passed to Drupal\Core\Entity\EntityViewBuilder::view() must implement interface Drupal\Core\Entity\EntityInterface, null given, called in /var/www/html/modules/oe_theme_content_consultation/oe_theme_content_consultation.module on line 102 in Drupal\Core\Entity\EntityViewBuilder->view() (line 133 of /var/www/html/build/core/lib/Drupal/Core/Entity/EntityViewBuilder.php).
```

So might be `if ($publication) {}` condition should be added.